src/kernel/lkm.rs

```diff
@@ -114,8 +114,13 @@ impl KernelModules<'_> {
 			SortType::Dependent => module_read_cmd += " | sort -n -r -t ' ' -k3",
 			_ => {}
 		}
-		let modules_content = util::exec_cmd("sh", &["-c", &module_read_cmd])
-			.expect("failed to read /proc/modules");
+		let modules_content = match util::exec_cmd("sh", &["-c", &module_read_cmd]) {
+			Ok(v) => v,
+			Err(e) => {
+				eprintln!("{e}");
```

Comment:
As I mentioned, the only problem with this approach is that you'll see the error message every time you press `r` (thus messes up the TUI) - so I suggest we don't print the error here.

We call this function for first time in `KernelModules::new` - maybe that's a good place to print the error before we even launch the TUI.

What do you think?

Reply:
You mean handling that in the mod.rs file?

Reply:
No I meant in the same file (`lkm.rs`), a little above. We can handle it in `mod.rs` too I guess. The point is we only print the error once.

Reply:
btw let me know if you need any guidance doing that - the codebase is kind of old and hard to figure out sometimes (I'm having issues myself lol)

Reply:
I tried to do this within `new()` function in `lmk.rs`, but couldn't figure out how should I do that, because we don't call module_content in there. What kind of ways should I consider?

Reply:
It turned out to be a bit more complicated than what I initially proposed, but we can simply apply this patch:


```diff
commit 06c0eb9dbd85e827d227964614ad6bdb3a7bab8f
Author: Orhun Parmaksız <orhunparmaksiz@gmail.com>
Date:   Sun Apr 7 00:01:16 2024 +0300

    refactor: Suppress refresh error

diff --git a/src/kernel/lkm.rs b/src/kernel/lkm.rs
index 3315003..c08bac6 100644
--- a/src/kernel/lkm.rs
+++ b/src/kernel/lkm.rs
@@ -5,6 +5,7 @@ use crate::util;
 use bytesize::ByteSize;
 use clap::ArgMatches;
 use ratatui::text::{Line, Span, Text};
+use std::error::Error;
 use std::slice::Iter;
 
 /* Type of the sorting of module list */
@@ -99,12 +100,14 @@ impl KernelModules<'_> {
 			args,
 			style,
 		};
-		kernel_modules.refresh();
+		if let Err(e) = kernel_modules.refresh() {
+			eprintln!("{e}");
+		}
 		kernel_modules
 	}
 
 	/* Parse kernel modules from '/proc/modules'. */
-	pub fn refresh(&mut self) {
+	pub fn refresh(&mut self) -> Result<(), Box<dyn Error>> {
 		let mut module_list: Vec<Vec<String>> = Vec::new();
 		/* Set the command for reading kernel modules and execute it. */
 		let mut module_read_cmd = String::from("cat /proc/modules");
@@ -114,13 +117,7 @@ impl KernelModules<'_> {
 			SortType::Dependent => module_read_cmd += " | sort -n -r -t ' ' -k3",
 			_ => {}
 		}
-		let modules_content = match util::exec_cmd("sh", &["-c", &module_read_cmd]) {
-			Ok(v) => v,
-			Err(e) => {
-				eprintln!("{e}");
-				String::new()
-			}
-		};
+		let modules_content = util::exec_cmd("sh", &["-c", &module_read_cmd])?;
 		/* Parse content for module name, size and related information. */
 		for line in modules_content.lines() {
 			let columns: Vec<&str> = line.split_whitespace().collect();
@@ -143,6 +140,8 @@ impl KernelModules<'_> {
 		self.default_list = module_list.clone();
 		self.list = module_list;
 		self.scroll_list(ScrollDirection::Top);
+
+		Ok(())
 	}
 
 	/**
diff --git a/src/kernel/mod.rs b/src/kernel/mod.rs
index dd55b33..e7b2446 100644
--- a/src/kernel/mod.rs
+++ b/src/kernel/mod.rs
@@ -34,6 +34,6 @@ impl Kernel {
 	pub fn refresh(&mut self) {
 		self.logs.refresh();
 		self.info.refresh();
-		self.modules.refresh();
+		let _ = self.modules.refresh();
 	}
 }
```

Here we still return the error, but print the error when `KernelModules` is constructed and suppress it for the `refresh` function.

Reply:
@orhun I applied your changes

Reply:
but as I said before I couldn't test it because I can't reproduce that kernel type

Reply:
that's fine, you can return a dummy error from that function (via `panic!`) just for testing 🙂